pkg/email/sender_test.go

```diff
@@ -0,0 +1,215 @@
+
+	assert.Nil(t, err)
+	assert.Contains(t, input.Body, "https://example.com/api/unsubscribe")
+	assert.Contains(t, input.Body, "2025-01-01")
+}
+
+func testGenerateBodyFromHTMLWeatherDaily(t *testing.T) {
+	t.Parallel()
+
+	cfg := testutils.SetupTestConfig(t)
+
+	input := &email.SendEmailInput{
+		To:      "test@example.com",
+		Subject: "Daily Email",
+	}
+	templateData := service.WeatherForecastDailyEmailTemplateInput{
+		UnsubscribeLink: "https://example.com/api/unsubscribe",
+		City:            "London",
+		Weather: domain.DayWeatherResponse{
+			SevenAM: domain.WeatherResponse{
+				Temperature: 18.0,
+				Humidity:    70,
+				Description: "Cloudy",
+			},
+			TenAM: domain.WeatherResponse{
+				Temperature: 18.0,
+				Humidity:    70,
+				Description: "Cloudy"},
+			OnePM: domain.WeatherResponse{
+				Temperature: 18.0,
+				Humidity:    70,
+				Description: "Cloudy"},
+			FourPM: domain.WeatherResponse{
+				Temperature: 18.0,
+				Humidity:    70,
+				Description: "Cloudy"},
+			SevenPM: domain.WeatherResponse{
+				Temperature: 18.0,
+				Humidity:    70,
+				Description: "Cloudy"},
+			TenPM: domain.WeatherResponse{
+				Temperature: 18.0,
+				Humidity:    70,
+				Description: "Cloudy"},
+		},
+		Date: "2025-01-01",
+	}
+
+	err := input.GenerateBodyFromHTML(cfg.Email.Templates.WeatherForecastDaily, templateData)
+
+	assert.Nil(t, err)
+	assert.Contains(t, input.Body, "https://example.com/api/unsubscribe")
+	assert.Contains(t, input.Body, "2025-01-01")
+}
+
+func testGenerateBodyFromHTMLInvalidTemplateFile(t *testing.T) {
+	t.Parallel()
+
+	input := &email.SendEmailInput{
+		To:      "test@example.com",
+		Subject: "Invalid Template",
+	}
+
+	// Non-existent file
+	err := input.GenerateBodyFromHTML("non_existent_template.html", nil)
+
+	assert.Error(t, err)
+	assert.Empty(t, input.Body)
+}
+
+func testGenerateBodyFromHTMLTemplateExecutionError(t *testing.T) {
+	t.Parallel()
+
+	// Create a temporary file with an invalid template (refers to a missing field)
+	tmpFile, err := os.CreateTemp("", "bad_template_*.html")
+	assert.NoError(t, err)
+	defer func(name string) {
+		err := os.Remove(name)
+		if err != nil {
+			t.Fatalf("failed to remove temp file %s: %v", name, err)
+		}
+	}(tmpFile.Name())
+
+	// Write template with invalid field (e.g., {{.MissingField}})
+	content := `<!DOCTYPE html><html><body>{{.MissingField}}</body></html>`
+	_, err = tmpFile.Write([]byte(content))
+	assert.NoError(t, err)
+	err = tmpFile.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	input := &email.SendEmailInput{
+		To:      "test@example.com",
+		Subject: "Bad Execution",
+	}
+
+	// Pass data that doesn't have `.MissingField`
+	err = input.GenerateBodyFromHTML(tmpFile.Name(), struct{ ValidField string }{ValidField: "hello"})
+
+	assert.Error(t, err)
+	assert.Empty(t, input.Body)
+}
+
+func TestSendEmailInput_Validate(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name    string
+		input   email.SendEmailInput
+		wantErr string
```

Comment:
```suggestion
		wantErr error
```

Можемо замінити? Тоді в тестах де помилка не очікується можна буде проставляти `wantErr: nil`
А сама перевірка буде:
```go
if c.wantErr == nil {
  assert.NoError(t, err)
} else {
  assert.ErrorIs(t, err, c.wantErr)
}
```

Reply:
згоден, пофіксив